### ISSUE TITLE
[data grid] Data Grid is failing AXE accessibility test with the message "Certain ARIA roles must contain particular children"

### ISSUE THREAD

**Issue by caseyhenderson** [mui/mui-x#13685]
### Steps to reproduce

Link to live example:  https://mui.com/x/react-data-grid/#pro-plan (Can be reproduced on the current MUI website)
And also in this demo reproduction
https://codesandbox.io/p/sandbox/aged-waterfall-86mxz6?file=%2Fsrc%2FDemo.tsx 

Pre-requirements
- AXE Dev Tools extension in your browser of choice.
Observed on Google Chrome 126.0.6478.127 (latest at time of writing)

Steps:
1. Navigate to a page containing a MUI Data Grid (any type)
2. Begin accessibility testing with the AXE Dev Tools: click 'start new scan' and then 'full page scan' when given the option.
3. You will notice that the AXE Dev Tools detect the issue described, which is classed as "critical"

### Current behavior

A critical accessibility issue "Certain ARIA roles must contain particular children" is detected when a full page scan is run using the AXE Dev Tools on any page containing the latest MUI Data Grid. 
![Screenshot 2024-07-01 124356](https://github.com/mui/mui-x/assets/61591930/76a43073-57d0-4e17-8a37-ad4396fb2614)

This means the Data Grid currently does not meet the WCAG 2.1 (A) standard - see here https://dequeuniversity.com/rules/axe/4.9/aria-required-children?application=AxeChrome.

### Expected behavior

This accessibility issue should not be present, as it was apparently addressed in https://github.com/mui/mui-x/pull/11428. As such, when the AXE Dev Tools are run against https://mui.com/x/react-data-grid/#pro-plan, no accessibility issues should be detected relating to the Data Grid.

### Context

This issue was previously addressed with the ariav7 experimental flag when using v6, and appeared to be fixed in MUI-X v7 in this PR https://github.com/mui/mui-x/pull/11428 . However, accessibility testing reveals this issue is still present on the latest version of V7. The problem here is that our application must be WCAG compliant, and so we cannot upgrade our DataGrids to V7 until this issue is resolved.

Notably this issue is also present in all public-facing demos of the DataGrid, including those on the MUI docs website itself. I have included a screenshot, and the issue can be easily observed by navigating to https://mui.com/x/react-data-grid/#pro-plan and running the AXE Dev Tools in Chrome.

More can be found on this accessibility issue here https://dequeuniversity.com/rules/axe/4.9/aria-required-children?application=AxeChrome. 

I've observed this on the latest MUI-X version, but also on v7.0, v7.4.0, and v7.5.0 (annoyingly I haven't been able to find a v7 release where this was working as expected) . If any further information is required I'd be more than happy to provide it - please get in touch!

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: accessibility, aria-roles, data-grid-pro
**Order ID**: 89221

**Comment by caseyhenderson** [comment]
No environment details included as I have been able to reproduce this on the live MUI website here https://mui.com/x/react-data-grid/#pro-plan

**Comment by romgrk** [comment]
Duplicate of #12837
